code/campaign/build.gradle.kts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+plugins {
+    id("aep-library")
+}
+
+val mavenCoreVersion: String by project
+val functionalTestIdentityVersion: String by project
+val functionalTestLifecycleVersion: String by project
+
+aepLibrary {
+    namespace = "com.adobe.marketing.mobile"
+    enableSpotless = true
+    enableCheckStyle = true
+
+    publishing {
+        gitRepoName = "aepsdk-campaign-android"
+        addCoreDependency(mavenCoreVersion)
+    }
+}
+
+dependencies {
+    implementation("com.adobe.marketing.mobile:core:$mavenCoreVersion-SNAPSHOT")
+
+    // testImplementation dependencies provided by aep-library:
+    // MOCKITO_CORE, MOCKITO_INLINE, JSON
+
+    // androidTestImplementation dependencies provided by aep-library:
+    androidTestImplementation("com.adobe.marketing.mobile:identity:$functionalTestIdentityVersion-SNAPSHOT")
```

Comment:
you may need to exclude core in Identity snapshot.  You might get error when download  the dependency for the app.  Same for Lifecycle.
androidTestImplementation("com.adobe.marketing.mobile:identity:$functionalTestIdentityVersion-SNAPSHOT")
     {
         exclude(group = "com.adobe.marketing.mobile", module = "core")
     }

Reply:
I think all the extensions in the functional test
- Identity
- Lifecycle
 uses core-3.0.0-SNAPSHOT build..
 I am not seeing any errors as of now building them. Will reconsider if I get any compile time errors